modules/gateway/conn.go

```diff
@@ -17,3 +17,14 @@ type peerConn struct {
 func (pc peerConn) RPCAddr() modules.NetAddress {
 	return pc.dialbackAddr
 }
+
+// dial will dial the input address and return a connection. dial appropriately
+// handles things like clean shutdown, fast shutdown, and chooses the correct
+// communication protocol.
+func (g *Gateway) dial(addr modules.NetAddress) (net.Conn, error) {
+	dialer := &net.Dialer{
+		Timeout: dialTimeout,
+		Cancel:  g.threads.StopChan(),
```

Comment:
whoa, does that work? can't believe I missed that :stuck_out_tongue_closed_eyes: